app/master.js

```diff
@@ -128,11 +133,20 @@ module.exports = function Master() {
     });
   };
   // Reload all fork workers and return Promise
-  const reloadAllWorkers = () => Promise
-    .each(_.values(cluster.workers), reloadWorker)
-    .then(() => {
-      logger.info('All workers restarted.');
-    });
+  const reloadAllWorkers = () => {
+    if (chainForks === false) {
+      return Promise
+        .each(_.values(cluster.workers), reloadWorker)
```

Comment:
this should do also forks one by one (http://bluebirdjs.com/docs/api/promise.each.html)